jwst/background/background_sub.py

```diff
@@ -295,7 +296,12 @@ def subtract_wfss_bkg(input_model, bkg_filename, wl_range_name, mmag_extract=Non
     # i.e. in regions we can use as background.
     if got_catalog:
         bkg_mask = mask_from_source_cat(input_model, wl_range_name, mmag_extract)
-        if bkg_mask.sum() < 100:
+        # Ensure mask has 100 pixels and that those pixels correspond to valid
+        # pixels using model DQ array
+        if np.count_nonzero(input_model.dq[bkg_mask]
+                            ^ pixel['DO_NOT_USE']
+                            & pixel['DO_NOT_USE']
```

Comment:
Not 100% sure what this new arrangement of operators is doing (first using XOR to find pixels that are not flagged as bad in the bkg_mask and then finding those that are bad in the regular dq mask?), but if it works, go for it!

Reply:
The first method used the modulo operator to get right to whether the 1 bit was on or not - to generalize to an arbitrary bit, these operators are:
> taking the dq bits of pixels deemed to be in the region defined as the background,
> flipping the DO_NOT_USE bit,
> checking to see if the DO_NOT_USE bit is on,
> then counts the pixels where this is the case.

in short, it counts how many background pixels have the DO_NOT_USE flag set to False/0. Because the optimal case where all pixels have "clean" dqflags reduces to counting the pixels, I'm fairly confident this if statement is a more general case of the previous statement and so the old sum can be safely removed.